File: gravitee-apim-definition/gravitee-apim-definition-model/src/main/java/io/gravitee/definition/model/federation/FederatedApi.java
/*
 * Copyright © 2015 The Gravitee team (http://gravitee.io)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.gravitee.definition.model.federation;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.gravitee.definition.model.ApiDefinition;
import io.gravitee.definition.model.DefinitionVersion;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import java.io.Serializable;
import java.util.Map;
import java.util.Set;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder(toBuilder = true)
public class FederatedApi implements Serializable, ApiDefinition {

    @JsonProperty(required = true)
    @NotBlank
    private String id;

    @JsonProperty(required = true)
    @NotBlank
    private String providerId;

    @JsonProperty(required = true)
    @NotBlank
    private String name;

    @JsonProperty(required = true)
    @NotBlank
    private String apiVersion;

    @JsonProperty(required = true)
    @NotNull
    @Builder.Default
    private DefinitionVersion definitionVersion = DefinitionVersion.FEDERATED;

    @JsonProperty(required = true)
    @NotBlank
    private Map<String, String> server;

    @Override
    public Set<String> getTags() {
        return Set.of();
    }

    @Override
    public void setTags(Set<String> tags) {
        // nothing to do
    }
}

File: gravitee-gamma/gravitee-gamma-module-apim/src/main/ui/features/apis/utils/federatedApi.ts
/*
 * Copyright © 2015 The Gravitee team (http://gravitee.io)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import type { ApiDetailDto, ApiListItem } from '../types';

export function isFederatedApi(api: ApiDetailDto | null | undefined): boolean {
    return api?.definitionVersion === 'FEDERATED';
}

export function isFederatedAgentApi(api: ApiDetailDto | null | undefined): boolean {
    return api?.definitionVersion === 'FEDERATED_AGENT';
}

// Keyed on the row's own `definitionVersion` rather than on `originContext.origin`, which a
// FEDERATED_AGENT row also carries as INTEGRATION and which therefore cannot tell the two apart.
export function isFederatedApiListItem(api: ApiListItem): boolean {
    switch (api.definitionVersion) {
        case 'FEDERATED':
            return true;
        case 'FEDERATED_AGENT':
        case 'V4':
        case 'V2':
            return false;
        default: {
            const unhandled: never = api.definitionVersion;
            return unhandled;
        }
    }
}
